src/d/actor/d_a_player_main.cpp

```diff
@@ -2525,7 +4374,94 @@ BOOL daPy_lk_c::changeDamageProc() {
+        mLavaGndChk.SetPos(&current.pos);
+        if (checkNoResetFlg0(daPyFlg0_HOVER_BOOTS)) {
+            offNoResetFlg0(daPyFlg0_HOVER_BOOTS);
+            return procFall_init(2, daPy_HIO_wallCatch_c0::m.field_0x54);
+        } else if (dVar11 < -30.1f) {
+            f32 dVar8 = speed.x + m3730.x + m36A0.x + m36B8.x;
+            f32 dVar9 = speed.z + m3730.z + m36A0.z + m36B8.z;
+            f32 dVar10 = std::sqrtf(dVar8 * dVar8 + dVar9 * dVar9);
+            if (dVar10 > 0.001f) {
+                s16 sVar3 = cM_atan2s(dVar8, dVar9);
+                dVar9 = cM_ssin(sVar3);
+                dVar8 = cM_scos(sVar3);
+                current.pos.y = current.pos.y - speed.y;
+                cXyz local_74;
+                cXyz local_68;
+                local_74.x = current.pos.x;
+                local_74.y = current.pos.y - 5.0f;
+                local_74.z = current.pos.z;
+                local_68.x = local_74.x - (50.0f * dVar9);
+                local_68.y = local_74.y;
+                local_68.z = local_74.z - (50.0f * dVar8);
+                mLinkLinChk.Set(&local_74, &local_68, this);
+                cM3dGPla* pcVar4;
+                if (dComIfG_Bgsp()->LineCross(&mLinkLinChk)) {
+                    pcVar4 = dComIfG_Bgsp()->GetTriPla(mLinkLinChk);
+                } else {
+                    pcVar4 = NULL;
+                }
+                dVar10 = mVelocity;
+                if ((dComIfGs_getLife() != 0) || (dComIfGs_checkBottle(dItem_FAIRY_BOTTLE_e))) {
+                    if ((m357C != 3) &&
+                        ((((!(checkGrabWear()) && (dVar10 >= daPy_HIO_autoJump_c0::m.field_0x14)) &&
+                           (mStickDistance > 0.85f)) &&
+                          (cLib_distanceAngleS(sVar3, shape_angle.y) < 0x800))))
+                    {
+                        if (pcVar4 != NULL) {
+                            current.pos.x = mLinkLinChk.GetCrossP()->x;
+                            current.pos.z = mLinkLinChk.GetCrossP()->z;
+                        }
+                        return procAutoJump_init();
+                    }
+                    if ((((pcVar4 != NULL) && (std::abs((pcVar4->mNormal).y) <= 0.05f)) &&
+                         ((dComIfG_Bgsp()->GetWallCode(mLinkLinChk) != 2 &&
+                           ((dVar11 < -125.0f && ((m35D0 - current.pos.y) < -125.0f)))))) &&
+                        (current.pos.y - m35D4 > 125.0f))
+                    {
+                        current.pos.x = mLinkLinChk.GetCrossP()->x;
+                        current.pos.z = mLinkLinChk.GetCrossP()->z;
+                        if (dComIfG_Bgsp()->GetWallCode(mLinkLinChk) == 1) {
+                            mPolyInfo = mLinkLinChk;
+                            return procClimbDownStart_init(
+                                cM_atan2s(pcVar4->mNormal.x, pcVar4->mNormal.z));
+                        }
+                        if (procHangFallStart_init(pcVar4) != 0) {
```

Comment:
This returns a boolean
```suggestion
                        if (procHangFallStart_init(pcVar4)) {
```